### PR TITLE
applications: nrf_desktop: Rework state handling in motion module

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-591

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

**Comment by pdunaj** [comment]
PR 1000 :)

**Comment by pdunaj** [comment]
tested with normal and lowlatency code using pca20041. seems working as usual no performance issues noticed.

File: applications/nrf_desktop/src/hw_interface/motion_sensor.c [hunk 1]
```diff
@@ -463,8 +484,13 @@ static bool event_handler(const struct event_header *eh)
 		switch (state.state) {
 		case STATE_FETCHING:
 		case STATE_IDLE:
+		case STATE_DISCONNECTED:
 			enable_trigger();
-			state.state = STATE_SUSPENDED;
+			if (state.state == STATE_DISCONNECTED) {
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/hw_interface/motion_sensor.c]: Maybe we could move the code around and remove if ?

File: applications/nrf_desktop/src/hw_interface/motion_sensor.c [hunk 2]
```diff
@@ -443,10 +459,15 @@ static bool event_handler(const struct event_header *eh)
 
 	if (is_wake_up_event(eh)) {
 		k_spinlock_key_t key = k_spin_lock(&state.lock);
-		if (state.state == STATE_SUSPENDED) {
+		if ((state.state == STATE_SUSPENDED) ||
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/hw_interface/motion_sensor.c]: Use switch case here ?

File: applications/nrf_desktop/src/hw_interface/motion_sensor.c (hunk 1 above)
Comment (pdunaj) [reply applications/nrf_desktop/src/hw_interface/motion_sensor.c]: I was thinking about it but decided not too. This is to not duplicate the code.

Comment (pdunaj) [reply applications/nrf_desktop/src/hw_interface/motion_sensor.c]: thought about it but decided not to to avoid duplication

**Comment by pdunaj** [comment]
Hi @jhn-nordic , @wiminordic , could you give a green tick here?

**Pull request merged**